kyo-core/shared/src/main/scala/kyo/Adder.scala

```diff
@@ -4,7 +4,7 @@ import java.util.concurrent.atomic as j
 
 /** A wrapper for Java's LongAdder.
   */
-class LongAdder private[kyo] (private val ref: j.LongAdder) extends AnyVal:
+final case class LongAdder private (unsafe: LongAdder.Unsafe) extends AnyVal:
```

Comment:
I tried that initially but it's problematic because the compiler has trouble to know which extension method should be used. The extensions for both the regular and unsafe versions become valid in the scope.